weaver/processes/wps1_process.py

```diff
@@ -112,108 +118,102 @@ def get_input_values(self, process, workflow_inputs):
-            job_id = "<undefined>"
-            while execution.isNotComplete() or run_step == 0:
-                if num_retries >= max_retries:
-                    raise Exception("Could not read status document after {} retries. Giving up.".format(max_retries))
-                try:
-                    execution = check_wps_status(location=execution.statusLocation,
-                                                 sleep_secs=wait_secs(run_step), settings=self.settings)
-                    job_id = execution.statusLocation.replace(".xml", "").split("/")[-1]
-                    LOGGER.debug(get_log_monitor_msg(job_id, status.map_status(execution.getStatus()),
-                                                     execution.percentCompleted, execution.statusMessage,
-                                                     execution.statusLocation))
-                    self.update_status(get_job_log_msg(status=status.map_status(execution.getStatus()),
-                                                       message=execution.statusMessage,
-                                                       progress=execution.percentCompleted,
-                                                       duration=None),  # get if available
-                                       map_progress(execution.percentCompleted,
-                                                    REMOTE_JOB_PROGRESS_MONITORING, REMOTE_JOB_PROGRESS_FETCH_OUT),
-                                       status.STATUS_RUNNING)
-                except Exception as exc:
-                    num_retries += 1
-                    LOGGER.debug("Exception raised: %r", exc)
-                    sleep(1)
-                else:
-                    num_retries = 0
-                    run_step += 1
-
-            if not execution.isSucceded():
-                exec_msg = execution.statusMessage or "Job failed."
-                LOGGER.debug(get_log_monitor_msg(job_id, status.map_status(execution.getStatus()),
-                                                 execution.percentCompleted, exec_msg, execution.statusLocation))
-                raise Exception(execution.statusMessage or "Job failed.")
-
-            self.update_status("Fetching job outputs from remote WPS1 provider.",
-                               REMOTE_JOB_PROGRESS_FETCH_OUT, status.STATUS_RUNNING)
-
-            results = [ows2json_output_data(output, process, self.settings) for output in execution.processOutputs]
-            for result in results:
-                result_id = get_any_id(result)
-                result_val = get_any_value(result)
-
-                # TODO Should we handle other type than File reference?
-                if result_id in expected_outputs:
-                    # This is where cwl expect the output file to be written
-                    # TODO We will probably need to handle multiple output value...
-                    dst_fn = "/".join([out_dir.rstrip("/"), expected_outputs[result_id]])
```

Comment:
It looks like you dropped the expected_outputs name usage. I saw the note in the `stage_results` function, but wonder if using the result name directly from wps is reliable?

Reply:
The same procedure is still done here: 
https://github.com/crim-ca/weaver/pull/372/files/31d5e80bced3d1cdf6a5462573a4cd2149f6f9b3#diff-96df54de7a27467cb71c60825abc6cca53a5c2030874273982ecef6a89843b34R270-R273

Expected outputs IDs are not needed during result retrieval from the WPS because the results generated here come from `execution.processOutputs`, which is generated from the WPS process definition itself.